### ISSUE TITLE
DuckieTV OSX Standalone - Not responding on fresh install

### ISSUE THREAD

**Issue by moswain** [SchizoDuckie/DuckieTV#1422]
**What build of DuckieTV are you using (Standlone / Chrome Extension (New Tab / Browser Action))**
...Standalone

**What version of DuckieTV are you using (Stable 1.1.x / Nightly yyyymmddHHMM)**
...DuckieTV-202404110853-OSX-x64

**What is your Operating System (Windows, MacOSx, Linux)**
...Sonoma 14.4.1 (23E224)

**Describe the problem you are having and steps to reproduce if available**
..."Application not responding" on a fresh install. Force Quit and restarting app + mac doesn't resolve the issue.

**Attach any DuckieTV statistics or Developer Console logs if available**
...
<img width="301" alt="Screenshot 2024-04-30 at 19 00 34" src="https://github.com/SchizoDuckie/DuckieTV/assets/106866058/a66c759f-76f9-4330-8af6-767013838e16">

**Comment by ilike2burnthing** [comment]
Do you have the same issue with older DuckieTV builds?

**Comment by moswain** [comment]
Yes, I’ve tried builds from 2024 Jan as well.

—
Moswain Antão

**Comment by ilike2burnthing** [comment]
I don't have an OSX device, but I've just tested it on https://tria.ge and it installed and ran without issue. Tried installing and running in safe mode?

**Comment by garfield69** [comment]
As you may know, Dtv is written in JavaScript using the AngularJS framework as a Chrome Browser extension.
The standalone flavour is exactly the same JS code, encapsulated by the [NWJS](https://github.com/nwjs/nw.js) run-time package, which includes a built in Chromium headless browser, so that Dtv can run as an app.

A quick troll through the web shows that Sonoma 14.4 introduced several major issues, including one that caused java processes to terminate unexpectedly. (See this article: https://www.macworld.com/article/2274944/macos-sonoma-14-4-update-bugs-usb-devices-icloud-drive-java.html)
While update 14.4.1 provided some fixes for many issues caused by 14.4 it may well be the case that not all java issues have been fixed.

Also its entirely possible that the OSx gatekeeper is interfering as the NWJS package that Dtv uses is unsigned, so that may cause complications downstream.

I don't have access to a mac OSx anymore, and my windows virtualbox no longer allows me to run an OSx image without a license, so I cannot provide any support for that platform. Sorry.

**Comment by moswain** [comment]
[Error log DTV .txt](https://github.com/SchizoDuckie/DuckieTV/files/15177562/Error.log.DTV.txt)

I tried opening the app in OSX safe mode. It's the same.

Would the MacOS error report help in anyway?

**Comment by ilike2burnthing** [comment]
Sure, might as well.

I don't know if DuckieTV has been tested on MacOS ARM before.

**Comment by garfield69** [comment]
Oh that an easy answer. The Dtv builds do not include a MacOS ARM build, there is only an OSx x64 build available.

**Comment by ilike2burnthing** [comment]
Having never looked at a MacOS error log before, do not take my word on this, but I _think_ that it was running with Rosetta, but I don't think that is a guarantee for success.

**Comment by moswain** [comment]
<img width="264" alt="Screenshot 2024-05-02 at 00 18 26" src="https://github.com/SchizoDuckie/DuckieTV/assets/106866058/e245a0fd-76b4-4452-827b-dfa6381127fe">

There isn't an option to open it with Rosetta 2.

**Comment by moswain** [comment]
so if CWe cannot use DuckieTV on Chrome, not standalone via ARM64 M1, M2, M3 macs.

Whats the best solution for mac users to use this app now?

**Comment by garfield69** [comment]
Rosetta works in the background. Once installed you do not need to select it, its automatic.
https://support.apple.com/en-us/102527

NWJS has recently added a mac osx arm package, https://dl.nwjs.io/v0.87.0/nwjs-sdk-v0.87.0-osx-arm64.zip
So if you are willing to try you could go the source run route.
See https://github.com/SchizoDuckie/DuckieTV/wiki/How-to-run-a-DuckieTV-Standalone-for-osX-ARM64
Good luck.
[edit] updated the links

**Comment by Js41637** [comment]
I'm a bit confused what the issue is here. I have a Macbook Pro with an M1 Max also on 14.4.1 (23E224) and the Duckie TV MacOS build seems to work fine. I had never tested it before but i just downloaded the OSX-x64.pkg and installed it and it opens and runs fine.

What macbook do you have? Does it perhaps work differently on an M2/M3 chip?

**Comment by garfield69** [comment]
FWIW, from the error log provided by the OP they appear to be using a MacBookAir10,1 which from https://support.apple.com/en-us/102869 is a M1 from the year 2020

**Comment by Js41637** [comment]
Well, there is definitely some issues. Sometimes when i exit the app and open it again it can become unresponsive for some reason. Seems like it can happen randomly sometimes as well, not sure... Also everytime it opens it asks you if you want to allow it to make connections on local network which is mildly annoying.

**Comment by garfield69** [comment]
@moswain  please try out the arm package at this temporary location https://github.com/SchizoDuckie/DuckieTV/releases/tag/1.1.6 and let us know how it goes.

**Comment by moswain** [comment]
I resorted to loading up a Win11 VM after your earlier comment, just to run DuckieTV and it was annoying!

This one worked like a Charm!!! Thank you!!! 
The app didn't start initially, I had to clear out the library folder below, but after that worked perfectly.
../Library/Application Support/DuckieTV-Standalone/

Points to note: 
It imported all my shows in seconds. Compared to previous instances where it took a while.
It did hang on importing the first time, but when I restarted the whole process, cleaned the library folder, and gave it another go it imported perfectly.

It's quite sluggish on the calendar tab. not aware if this is just this version or all versions.
It was quite smooth on the Chrome browser. All other tabs work smoothly and are responsive.

Will this arm package become part of the stable releases? 
Im sure a lot of users will benefit from it now that the Chrome extension has died.

I can keep at it for a few days and report back any bugs I notice.
